### PR TITLE
Fix/128807481 rma duplicate line item id quantity

### PR DESCRIPTION
**Pull request by poulping**
Hi, @hugobast this PR fixes the issue of duplicated line items in the RMARequest document.
Also, a failing shipment related spec has been updated, and useless test doubles have been removed. 

Merging this means we can delete this one https://github.com/DynamoMTL/gentle/pull/2

### CODE REVIEW

File: lib/gentle/documents/request/rma_document.rb [hunk 1]
```diff
@@ -31,11 +31,11 @@ def to_xml
...
                       'TrackingNumber' => @rma.tracking_number) do
 
-                @rma.returned_items.each do |returned_item|
-                  xml.Line('LineNo'          => returned_item.line_item_id,
+                @rma.returned_items.each_with_index do |returned_item, line_number|
+                  xml.Line('LineNo'          => line_number + 1,
                            'OrderNumber'     => @order.number,
                            'ItemNumber'      => returned_item.sku,
-                           'Quantity'        => returned_item.quantity,
+                           'Quantity'        => 1,
```
Comment (hugobast) [inline lib/gentle/documents/request/rma_document.rb:38]: Quantity is never anything other than 1?

**Comment by hugobast** [comment]
I'm not saying it's wrong and I am missing a lot of context but I feel like quantity always be 1 feels wrong.

**Comment by DanielWright** [comment]
In Spree/Solidus, return-items in RMAs are treated differently than line-items in shipment orders. They always have a quantity of one—basically, the customer is requesting to return each _inventory-unit_, not the line-item that was fulfilled by the inventory-units.

A more extensive explanation can be found [in Pivotal](https://www.pivotaltracker.com/story/show/128807481/comments/147625713).

**Comment by hugobast** [comment]
👍

**Comment by poulping** [comment]
Yes, @hugobast basically what Daniel said, the easy way to see this is with the pry in the ticket Dan mentioned.

**Pull request merged**